src/managers/GuildMemberManager.js

```diff
@@ -135,6 +136,50 @@ class GuildMemberManager extends BaseManager {
     return this._fetchMany(options);
   }
 
+  /**
+   * Edits a member of the guild.
+   * <info>The user must be a member of the guild</info>
+   * @param {UserResolvable} user The member to edit
+   * @param {GuildMemberEditData} data The data to edit the member with
+   * @param {string} [reason] Reason for editing this user
+   * @returns {Promise<void>}
+   */
+  async edit(user, data, reason) {
+    const id = this.client.users.resolveID(user);
+    if (!id) throw new TypeError('INVALID_TYPE', 'user', 'UserResolvable');
+
+    // Clone the data object for immutability
+    const _data = { ...data };
+    if (_data.channel) {
+      _data.channel = this.guild.channels.resolve(_data.channel);
+      if (!_data.channel || _data.channel.type !== 'voice') {
+        throw new Error('GUILD_VOICE_CHANNEL_RESOLVE');
+      }
+      _data.channel_id = _data.channel.id;
+      _data.channel = undefined;
+    } else if (data.channel === null) {
+      _data.channel_id = null;
+      _data.channel = undefined;
+    }
+    if (_data.roles) _data.roles = _data.roles.map(role => (role instanceof Role ? role.id : role));
+    let endpoint = this.client.api.guilds(this.guild.id);
+    if (id === this.client.user.id) {
+      const keys = Object.keys(_data);
+      if (keys.length === 1 && keys[0] === 'nick') endpoint = endpoint.members('@me').nick;
+      else endpoint = endpoint.members(id);
+    } else {
+      endpoint = endpoint.members(id);
+    }
+    await endpoint.patch({ _data, reason }).then(d => {
+      if (this.cache.has(id)) {
+        const member = this.cache.get(id);
+        const clone = member._clone();
+        d.user = member.user;
+        clone._patch(d);
```

Comment:
So if based on the above discussion we don't get any data, why patching with.. no data?

Reply:
Looked into the original again - it actually patches the GuildMember with the _data object, rather than any returned value.